src/onnx/parse_if.cpp

```diff
@@ -65,13 +67,108 @@ struct parse_if : op_parser<parse_if>
 
         auto then_out_shapes = then_mdl->get_output_shapes();
         auto else_out_shapes = else_mdl->get_output_shapes();
-        if(not std::equal(then_out_shapes.begin(),
-                          then_out_shapes.end(),
-                          else_out_shapes.begin(),
-                          else_out_shapes.end()))
-        {
+
+        auto throw_shapes = [&]() {
             MIGRAPHX_THROW("PARSE_IF: " + info.name +
-                           " then and else sub_grahps must have same output shapes!");
+                           " then and else sub_graphs must have compatible shapes ");
+        };
+
+        if(then_out_shapes.size() != else_out_shapes.size())
+        {
+            throw_shapes();
```

Comment:
Before we check for compatible shapes we should check that the condition is const, and then just inline the branch that is picked.

Reply:
Inline as in just use that and ignore the if creation all together?

Reply:
Yes, we used to do that:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/e00479af6dc513d8f83e6da773014dee51ce1c2c/src/onnx/parse_if.cpp#L60-L81

We removed it when we added the inliner(in #802), but it seems like it is still better to do inlining in the parser to handle more cases.

Reply:
Ah this seems to be nice if the IF is constant and would speed up our evaluation of this. I can add this in.

Reply:
Looks like this will break a bunch of tests if I do it right now. I'll add this change to the end or even a separate PR since I think we'll need additional verification test changes as well to ensure we get the right value.